patch to fix build post the glib2 update.  

Drop this once glade update to 3.22.1 is done (once TPNO is approved)


--- a/gladeui/glade-signal-model.c	2019-12-26 16:01:44.239876066 -0800
+++ b/gladeui/glade-signal-model.c	2019-12-26 16:09:16.262336231 -0800
@@ -63,18 +63,6 @@
                          G_IMPLEMENT_INTERFACE (GTK_TYPE_TREE_DRAG_SOURCE,
                                                 gtk_tree_drag_source_iface_init))
 
-static gint
-g_ptr_array_find (GPtrArray *array, gpointer data)
-{
-  gint i;
-  for (i = 0; i < array->len; i++)
-    {
-      if (array->pdata[i] == data)
-	return i;
-    }
-  return -1;
-}
-
 static void
 glade_signal_model_init (GladeSignalModel *object)
 {
@@ -525,11 +513,9 @@
 	    {
 	      if (handlers)
 		{
-		  gint handler_index = g_ptr_array_find (handlers, handler);
-		  if (handler_index == -1) /* dummy handler */
-		    {
-		      index1 += handlers->len;
-		    }
+		  guint handler_index; 
+                  if (g_ptr_array_find (handlers, handler, &handler_index))
+		    index1 += handlers->len;
 		  else
 		    index1 += handler_index;
 		}
@@ -583,7 +569,8 @@
         {
           GPtrArray *handlers = g_hash_table_lookup (sig_model->priv->signals,
                                                      glade_signal_get_name (signal));
-          if (!handlers || !handlers->len || g_ptr_array_find (handlers, signal) == 0)
+          guint index;
+          if (!handlers || !handlers->len || g_ptr_array_find (handlers, signal, &index) && index == 0)
             g_value_set_boolean (value, TRUE);
           else
             g_value_set_boolean (value, FALSE);
@@ -719,7 +706,10 @@
 	}
       else if (handlers)
 	{
-	  gint new_index = g_ptr_array_find (handlers, handler) + 1;
+	  gint new_index = 0;
+          if (g_ptr_array_find (handlers, handler, &new_index))
+            new_index++;
+
 	  if (new_index < handlers->len)
 	    {
 	      glade_signal_model_create_signal_iter (sig_model, widget,
